### DISCUSSION TITLE
Zabbix active agent

### DISCUSSION THREAD

**Discussion by pramod-desai-minutus** [apache/skywalking#10325] [Show and tell]
Hi !!
I am new to skywalking and want to monitor  linux host with zabbix rules in skywalking,

I have configured zabbix active agent , config looks like 

ServerActive=XX.XX.XX.XX:10051
Hostname=Linux 
RefreshActiveChecks=120

zabbix agent restarts successfully but host is not visible in skywalking UI.

By checking agent log file I'm getting following errors:

`cannot connect to [XX.XX.XX.XX:10051]: dial tcp :0->XX.XX.XX.XX:10051: connect: connection refused`

`sending of heartbeat message for [minutus-linux] started to fail`

` active check configuration update from host [Linux] started to fail`

Any suggestions with sample documentation woule be much appreciated.

Thanks in advance !!

**Comment by wu-sheng** [comment]
Do you activate OAP zabbix receiver as docs indicate?

**Reply by pramod-desai-minutus** [reply]
Yes,

receiver-zabbix:
  selector: ${SW_RECEIVER_ZABBIX:default}
  default:
    # Export tcp port, Zabbix agent could connected and transport data
    port: 10051
    # Bind to host
    host: 0.0.0.0
    # Enable config when receive agent request
    activeFiles: agent

I've configured this under **$CLASSPATH/zabbix-rules/agent.yaml**
![Screenshot (197)](https://user-images.githubusercontent.com/98507485/215756396-38e7def4-35d7-45ad-9a64-364883657c68.png)

After this configuration UI is not loading,by checking oap log file I'm getting

`Unable to find property 'receiver-zabbix' on class: org.apache.skywalking.oap.server.receiver.zabbix.provider.config.ZabbixConfig`

Can you help me with where to configire zabbix receiver correctly

Thank you!!

**Reply by wu-sheng** [reply]
Ping @mrproliu

**Reply by pramod-desai-minutus** [reply]
Sure, Thanks a lot!!

**Reply by mrproliu** [reply]
Once the OAP is started, is the `10051` port accessible for connections? If not, the configuration may be incorrect, possibly due to an incorrect format. Can you please supply more information regarding the log, configuration, and OAP version?

**Reply by mrproliu** [reply]
The OAP startup of the Zabbix plugin has been successful. But does the OAP or the Zabbix haven't any error log(I only saw the connect failure in the Zabbix)?

**Reply by mrproliu** [reply]
Ok, the error message shows the agent sending different protocol data. The OAP server only supports the communications protocol, please check your configuration of the Zabbix agent. Here is the official [Zabbix agent protocol](https://www.zabbix.com/documentation/6.2/en/manual/appendix/protocols/header_datalen), I think the agent side has compressed the data content.

**Reply by pramod-desai-minutus** [reply]
Worked , Thank you!!

**Comment by bangyogesh** [comment]
@mrproliu , @wu-sheng , Appreciate any pointers on how to report custom zabbix metrics (called items in Zabbix terminology) to Zabbix-Receiver under SKywalking ?
The custom metrics is defined in following way inside _agent.conf_ file of active zabbix-agent.
_UserParameter=mycustommetrics.mytime, date +%s_
